rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java

```diff
@@ -70,9 +77,11 @@
         }
 
         if (sqlRecognizer == null) {
+            boolean isSelectTransformForUpdate = (RootContext.requireGlobalLock() && CLIENT_RM_GLOBAL_LOCK_SELECT_FOR_UPDATE_ENABLE)
```

Comment:
How about change to
```
if (sqlRecognizer == null) {
                statementProxy.getTargetSQL(),
                statementProxy.getConnectionProxy().getDbType(),
                    CLIENT_RM_GLOBAL_LOCK_SELECT_FOR_UPDATE_ENABLE || CLIENT_RM_GLOBAL_TRANSACTION_SELECT_FOR_UPDATE_ENABLE);
        }
```

Reply:
The mean is change this three row code can solve select automatically adds for update?
If that, I try.
Thanks

Reply:
![image](https://user-images.githubusercontent.com/45086633/78785375-a4532d00-79d9-11ea-9349-0bf4835114ab.png)
this is the master code